packages/trader/src/AppV2/Components/TradeParameters/RiskManagement/take-profit-and-stop-loss-input.tsx

```diff
@@ -0,0 +1,328 @@
+            if (
+                isMounted() &&
+                proposal &&
+                echo_req.contract_type === CONTRACT_TYPES.MULTIPLIER.UP &&
+                (echo_req?.limit_order?.take_profit || echo_req?.limit_order?.stop_loss)
+            ) {
+                const { validation_params } = proposal as ExpandedProposal;
+                setInfo(info => {
+                    if (
+                        (info.min_value !== validation_params?.[type]?.min && validation_params?.[type]?.min) ||
+                        (info.max_value !== validation_params?.[type]?.max && validation_params?.[type]?.max)
+                    ) {
+                        return {
+                            min_value: validation_params?.[type]?.min,
+                            max_value: validation_params?.[type]?.max,
+                        };
+                    }
+                    return info;
+                });
+            }
+            is_be_response_received_ref.current = true;
+        };
+
+        /* In order to get validation params for Multipliers when TP and SL are empty, 
+            we send '1' first, get validation params and set them into the state.*/
+        const input_value = should_set_validation_params ? '1' : new_input_value;
+        const dispose = requestPreviewProposal(
+            trade_store,
+            onProposalResponse,
+            {
+                ...(is_take_profit_input ? { has_take_profit: is_enabled } : { has_stop_loss: is_enabled }),
+                ...(is_take_profit_input
+                    ? { take_profit: is_enabled ? input_value : '' }
+                    : { stop_loss: is_enabled ? input_value : '' }),
+            },
+            true
+        );
+
+        return () => {
+            dispose?.();
+            WS.forget(subscription_id_ref.current);
+            clearTimeout(focus_timeout.current);
+        };
+
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [is_enabled, new_input_value]);
+
+    // TODO: remove all regex replacement, when comma will be supported by quill component with type number
+    const onInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        is_be_response_received_ref.current = false;
+        let value = String(e.target.value)
+            .replace(',', '.')
+            .replace(/[.](?!\d*$)/g, '');
+        if (value.startsWith('.')) {
+            value = value.replace(/^\./, '0.');
+        }
+        if (value.length > 1) {
+            value = /^[0-]+$/.test(value) ? '0' : value.replace(/^0*/, '').replace(/^\./, '0.');
+        }
+
+        setNewInputValue(value);
+        updateParentRef({ field_name: type, new_value: value });
+    };
+
+    const onSave = () => {
+        // Prevent from saving if user clicks before BE validation
+        if (!is_be_response_received_ref.current && is_enabled) return;
+        WS.forget(subscription_id_ref.current);
```

Comment:
is this relevant for Take profit as a standalone param, e.g. in Accumulators?

Reply:
Yes